doc/news/changes/minor/20170314DenisDavydov

```diff
@@ -0,0 +1,4 @@
+New: extend internal::VectorOperations::parallel_for() and
+internal::VectorOperations::parallel_reduce() to [start,end).
+<br>
+(Denis Davydov, 2017/03/14)
```

Comment:
@kronbichler i would prefer not to squash because the first commit is small, the second touches a lot of things, but should be easy. In case something gets broken it's good to keep them apart.